File: hdata-core/src/test/java/io/jayer/hdata/core/BeamDemo.java
package io.jayer.hdata.core;

import org.apache.beam.sdk.Pipeline;
import org.apache.beam.sdk.io.jdbc.JdbcIO;
import org.apache.beam.sdk.options.PipelineOptions;
import org.apache.beam.sdk.options.PipelineOptionsFactory;
import org.apache.beam.sdk.transforms.Create;
import org.apache.beam.sdk.transforms.ParDo;
import org.apache.beam.sdk.values.Row;

/**
 * @author Jayer
 * @date 2019-07-25
 */
public class BeamDemo {

    public static void main(String[] args) {
        JdbcIO.DataSourceConfiguration dataSourceConfiguration = JdbcIO.DataSourceConfiguration.create("com.mysql.cj.jdbc.Driver", "jdbc:mysql://10.247.17.31:3306/wuya_test?useSSL=false").withUsername("51generalnew").withPassword("uUZPeL32FpatOvju");

        PipelineOptions options = PipelineOptionsFactory.fromArgs(args).create();
        options.setJobName("Beam Test");
        Pipeline pipeline = Pipeline.create(options);
        pipeline.apply(Create.of((Void) null))
                .apply(ParDo.of(new JdbcReadSplittableRowDoFn()))
                .apply(JdbcIO.<Row>write()
                        .withDataSourceConfiguration(dataSourceConfiguration)
                        .withStatement("INSERT IGNORE INTO `T_Test_2` (`AutoId`, `Day`) VALUES (?, ?)")
                        .withPreparedStatementSetter((row, query) -> {
                                    query.setLong(1, row.getInt64(0));
                                    query.setString(2, row.getString(1));
                                }
                        )
                );

        pipeline.run().waitUntilFinish();
    }
}

File: hdata-jdbc/src/main/kotlin/io/jayer/hdata/jdbc/BeamDemo.kt
package io.jayer.hdata.jdbc

import org.apache.beam.sdk.Pipeline
import org.apache.beam.sdk.options.PipelineOptionsFactory
import java.util.*


/**
 * @author wuya
 * @date 2022-07-27
 */
object BeamDemo {

    @JvmStatic
    fun main(args: Array<String>) {
        val dataSourceConfig = Properties().apply {
            setProperty("jdbcUrl", "jdbc:mysql://10.247.17.31:3306/wuya_test?useSSL=false")
            setProperty("dataSource.user", "51generalnew")
            setProperty("dataSource.password", "uUZPeL32FpatOvju")
        }

        val sourceDescriptor = JdbcSourceDescriptor(
            dataSourceConfig,
            table = "T_Test_1",
            fetchSize = 50000
        )

        val sinkDescriptor = JdbcSinkDescriptor(dataSourceConfig, "T_Test_2", 3377)

        val options = PipelineOptionsFactory.fromArgs(*args).create()
        options.jobName = "Beam Test"
        val pipeline = Pipeline.create(options)
        pipeline.apply(JdbcStructuredSource(sourceDescriptor)).apply(JdbcStructuredSink(sinkDescriptor))
        pipeline.run().waitUntilFinish()
    }
}